CONTRIBUTING.md

```diff
@@ -50,6 +50,63 @@ Step 4. The Quarkus jdt.ls extension receives the command, determines
 information about the currently opened Quarkus project and returns
 the information to vscode-quarkus.  
 
+The information is computed by scanning all classes annotated with `@ConfigRoot` Quarkus annotation from all project classpath JARs and generating the proper Quarkus properties.
+
+For instance if the project has a `pom.xml` with this dependency:
+
+```xml
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-core-deployment</artifactId>
+</dependency>
+```
+
+The scanner will collect the [io.quarkus.deployment.ApplicationConfig](https://github.com/quarkusio/quarkus/blob/master/core/deployment/src/main/java/io/quarkus/deployment/ApplicationConfig.java) class annotated with `@ConfigRoot` and will generate a Quarkus property for each field:
+
+ * `quarkus.application.name` for the `name` field.
+ * `quarkus.application.version` for the `version` field.
+ 
+In addition, the Quarkus jdt.ls extension is able to manage `Quarkus deployment JAR`. 
+
+For instance if the project has a `pom.xml` with this dependency:
+
+```xml
+<dependency>
+    <groupId>io.quarkus</groupId>
+    <artifactId>quarkus-hibernate-orm</artifactId>
+</dependency>
+```
+
+this JAR doesn't contain the classes annotated with `@ConfigRoot`. The JAR which contains those classes is `quarkus-hibernate-orm-deployment*.jar`. This information comes from 
+the `META-INF/quarkus-extension.properties` of the `quarkus-hibernate-orm*.jar` as a property:
+		 
+```
+deployment-artifact=io.quarkus\:quarkus-hibernate-orm-deployment\:0.21.1
+```
+
+The downloads for deployment JARs are done with Maven. In other words, a project which declares:
```

Comment:
Under the hood, the quarkus jdt.ls extension delegate the resolution (i.e. download) of those deployment jars to Maven, regardless of the user project's build system. So, a project depending on: